lib/image/eight-bit-rgba-to-png.js

```diff
@@ -0,0 +1,80 @@
+const {unsigned: crc32} = require('buffer-crc32');
+const zlib = require('node:zlib');
+const { PNG } = require('../constants');
+
+const COMPRESSION_LEVEL = 3;
+
+const convertRgbaToScanlines = (rgba, width, height) => {
+    const stride = width * PNG.RGBA_CHANNELS;
+    const scanlines = Buffer.allocUnsafe(height * (1 + stride)); // extra byte for filter
+
+    let scanlineOffset = 0;
+    let pixelDataOffset = 0;
+
+    for (let y = 0; y < height; y++) {
+        scanlines.writeUInt8(PNG.FILTER_NO_FILTER, scanlineOffset);
+
+        rgba.copy(scanlines, scanlineOffset + 1, pixelDataOffset, pixelDataOffset + stride);
+
+        scanlineOffset += stride + 1;
+        pixelDataOffset += stride;
+    }
+
+    if (scanlineOffset !== scanlines.byteLength) {
+        throw new Error('Got malformed input while trying to convert rgba to png');
+    }
+
+    return scanlines;
+}
+
+exports.convertRgbaToPng = (rgba, width, height, compressionLevel = COMPRESSION_LEVEL) => {
```

Comment:
Just formatting data to https://www.w3.org/TR/PNG-Chunks.html standards
8 bit rgba deflate png without filter or interlace